### ISSUE TITLE
Websocket not reliable with Node-Red

### ISSUE THREAD

**Issue by rucksman** [gotify/server#438]
I know that you are not responsible für Node-Red support, nevertheless you probably can help.

I managed to get Gotify to run within Node-Red. Just for demo purposes I created a "websocket in" node which simply delivers its content to a debug-Node as soon as a message arrives. My "websocket client" node points to wss://gotify.myserver.tld/stream?token=myClientToken.

Immediately after deploying this simply flow it works. If I push a message from my smartphone, it is immediately shown in Node-Red in the debug node. But after some time ist stops working. After deploying the "websocket in" node says "connected", but afzer a certain timespan this "connected" disappears. And if I push a message from my smartphone, nothing is shown in Node-Red. I do not know after which timeframe this behaviour appears. Last time I checked it was yesterday evening, and today noon it did not work anymore. 

Is there kind of a timeout? Or is it a problem that my Gotify server is working on a hosted vserver in an external datacenter whereas my Node-Red machine is running at home, and every 24 hours I get a new IP from my internet provider?

**Comment by jmattheis** [comment]
Show your gotify/server log, in there should be a message why the websocket connection broke.

**Comment by rucksman** [comment]
The only entries in the log I can see which are not message or website related are:
`WebSocket: ReadError read tcp 172.18.0.2:80->172.18.0.3:57562: i/o timeout`
172.18.0.2 is my Gotify Docker container
172.18.0.3 is my NGINX proxy Docker container
Unfortunately these entries do not habe a timestamp, but when I look at the entry before and the next entry, these errors happen during the night when I get a new IP from my provider. For whatever reason I can see such entries also during the day. As soon as such an entry appears, the flow stops to work, although I just discovered that the "websocket in" node still says "connected". Then I change some minor thing in the "websocket client" node, and after deploying the flow starts to work again. Also there is an entry in the logs then:
`[GIN] 2021/09/30 - 09:46:32 | 200 |    6.422015ms |    XXX.XXX.XXX.XXX | GET      "/stream?token=myClientToken"`

**Comment by jmattheis** [comment]
Yeah, you ip switch is most likely the fault. Node-Red has to reconnect if the websocket connection breaks when the ip switches.

**Comment by rucksman** [comment]
The smartphone app does this reconnect automatically? I found this entry in the logs of the Nginx proxy which is realted to the token I use in Node-Red. Is appears every day at around the same time:
`nginx.1     | gotify.myserver.tld XXX.XXX.XXX.XXX - - [01/Oct/2021:05:06:17 +0200] "GET /stream?token=myClientToken HTTP/1.1" 101 3704 "-" "-" "172.18.0.2:80"`

**Comment by jmattheis** [comment]
Yes, the android app has auto reconnect.
